src/q7/cl-INGI2263/summary2/chapters/cm34.tex

```diff
@@ -0,0 +1,281 @@
+\end{enumerate}
+The initialization step is only performed once at the beginning.
+The EM-step is performed several times, until convergence.
+
+\subsubsection{Backoff Smoothing}
+\textbf{Backoff smoothing} is based on the idea of \textbf{smoothing by discounting}:
+\[
+\Prhat(w \mid h) = \frac{C(h, w) - d_C}{C(h)}, \quad \textnormal{if} \quad C(h, w) > 0.
+\]
+By discounting a certain value \(d_C\) (which can depend on the value of \(C(h, w)\)), we want to compensate for the \emph{over-estimated probability} of \emph{observed} \(N\)-grams (\(C(h, w) > 0\)).
+The \emph{discounted probability mass} will be used to compensate for the \emph{under-estimated probability} of \emph{non-observed} \(N\)-grams in the training corpus, as without smoothing
+\[
+C(h, w) = 0 \implies \Prhat(w \mid h) = \frac{C(h, w)}{C(h)} = 0.
+\]
+
+\paragraph{Good--Turing discounting} One of the methods to compute the \emph{discounting coefficients} \(d_C\) deduces their values from performing ML estimation on \emph{leave-one-out} (LOO) samples (\(N\)-grams appearing once in the training set then become unobserved).
+The \textbf{Good--Turing} estimate derives from this principled approach:
+\[
+C - d_C = (C+1) \frac{n_{C+1}}{n_C},
+\]
+where \(n_C\) is the number of \(N\)-grams occurring \(C\) times.
+Note that this value depends on the \emph{model order}.
+When \(C\) increases, the ratio \(n_{C+1}/n_C\) can become greater than \(1\) (to see this, consider the count histograms mentioned earlier).
+For this reason, Good--Turing discounting is often limited to small values of \(C\), with ML estimates being used for larger values of \(C\).
+
+\paragraph{Absolute discounting} Another method makes the assumption that discounting is \emph{independent} of the value of \(C\): \(d_C \triangleq d_*\), for all \(C\).
+An optimal \textbf{absolute discounting} coefficient can be estimated from \(n_C\), the number of \(N\)-grams appearing \(C\) times:
+\[
+d_* = \frac{n_1}{n_1 + 2n_2}.
+\]
+In general, \(n_1, n_2 > 0\), hence \(0 < d_* < 1\).
+As for Good--Turing discounting, the discounted value depends on the \emph{model order}.
+This discounting strategy has the (nice) property that it will affect low counts relatively more heavily than higher counts, which makes sense as the low count \(N\)-grams are the ones most likely to be assigned an overestimated probability.
+
+We can then also define a \emph{total discounted probability mass}, which, for a given history \(h\), is
+\[
+\gamma(h) = \sum_{\{w \in \mathcal{W}: C(h, w) > 0\}} \frac{d_C}{C(h)},
+\]
+and should be redistributed over the unobserved \(N\)-grams.
+
+\paragraph{Katz Backoff Smoothing} One way of redistributing \(\gamma(h)\) is given by the \textbf{Katz backoff smoothing}:
+\[
+\Prhat(w \mid h) = \left\{\begin{array}{l@{\quad}l}
+\frac{C(h, w) - d_C}{C(h)}, & \textnormal{if} \quad C(h, w) > 0,\\
+\frac{1}{Z} \gamma(h) \Prhat_{\textnormal{backoff}}(w \mid h), & \textnormal{if} \quad C(h, w) = 0,
+\end{array}\right.
```

Comment:
Just wondering, is there a reason for not using `\begin{cases} ... \end{cases}` ?